src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -51,5 +51,23 @@ protected override bool IsSystemNamespaceImported(IReadOnlyList<SyntaxNode> impo
             }
             return false;
         }
+
+        protected override bool IsPassingZeroAndBufferLength(SyntaxNode bufferValueNode, SyntaxNode offsetValueNode, SyntaxNode countValueNode)
+        {
+            return
+                // First argument should be an identifier name node
+                bufferValueNode is IdentifierNameSyntax firstArgumentIdentifierName &&
+                // Second argument should be a literal expression node...
+                offsetValueNode is LiteralExpressionSyntax secondArgumentLiteralExpression &&
+                // and its value should be zero
+                secondArgumentLiteralExpression.Token.ValueText == "0" &&
```

Comment:
@mavasani is this the proper way to check if the value of the token is exactly `0`? Same question for line 70 to check for `Length`.

Reply:
A better approach is to check for constant value for the syntax using this API: http://sourceroslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,743

Reply:
You can also remove the check `offsetValueNode is LiteralExpressionSyntax secondArgumentLiteralExpression`, instead just check constant value of syntax node for second argument.

Reply:
So it seems I would change my code to something like this:
```suggestion
                model.GetConstantValue(secondArgumentLiteralExpression) is Optional<object> optionalObject &&
                is Optional<object> optionalValue && optionalValue.HasValue &&
                optionalValue.Value is string value && value.Length == 0 &&
```

But I'm wondering why would that be an improvement? Wouldn't all those additional comparisons and casts have a performance impact?

Reply:
You will not have `optionalValue.Value is string value && value.Length == 0` but instead `optionalValue.Value is int value && value == 0`. The code allows for cases were instead of `0` you used some constant field assigned to `0` and also feels much cleaner then explicitly checking for syntax token `0`. Its fine even to retain the current code if you feel this case is not too important.

Reply:
> _optionalValue.Value is string value && value.Length == 0_

Sorry, I pasted the wrong text. I was looking for usage examples of your suggestion and found one with those parameters. I then adapted it to my code, but I didn't realize that I ended up pasting here the code from the other example. I edited my auto-suggestion.

But the general question still applies: the fact that I'm doing more castings and comparisons that directly checking ValueText.

Maybe adding the extra casts is not that impactful, and I think it's important to call the right code. I don't know if comparing strings could cause trouble if the text has an unexpected property, like an unusual encoding.